integration/loki_micro_services_delete_test.go

```diff
@@ -129,66 +195,92 @@ func TestMicroServicesDeleteRequest(t *testing.T) {
 	t.Run("query again to verify logs being served from storage", func(t *testing.T) {
 		resp, err := cliQueryFrontend.RunRangeQuery(context.Background(), `{job="fake"}`)
 		require.NoError(t, err)
-		assert.Equal(t, "streams", resp.Data.ResultType)
+		validateQueryResponse(resp)
+	})
 
-		var lines []string
-		for _, stream := range resp.Data.Stream {
-			for _, val := range stream.Values {
-				lines = append(lines, val[1])
+	t.Run("add-delete-requests", func(t *testing.T) {
+		for _, deleteRequest := range expectedDeleteRequests {
+			params := client.DeleteRequestParams{
+				Start: strconv.FormatInt(deleteRequest.StartTime, 10),
+				End:   strconv.FormatInt(deleteRequest.EndTime, 10),
+				Query: deleteRequest.Query,
 			}
+			require.NoError(t, cliCompactor.AddDeleteRequest(params))
 		}
-
-		assert.ElementsMatch(t, []string{"lineA", "lineB", "lineC", "lineD"}, lines)
-	})
-
-	t.Run("add-delete-request", func(t *testing.T) {
-		params := client.DeleteRequestParams{Start: "0000000000", Query: `{job="fake"} |= "lineB"`}
-		require.NoError(t, cliCompactor.AddDeleteRequest(params))
 	})
 
 	t.Run("read-delete-request", func(t *testing.T) {
 		deleteRequests, err := cliCompactor.GetDeleteRequests()
 		require.NoError(t, err)
-		require.NotEmpty(t, deleteRequests)
-		require.Len(t, deleteRequests, 1)
-		require.Equal(t, `{job="fake"} |= "lineB"`, deleteRequests[0].Query)
-		require.Equal(t, "received", deleteRequests[0].Status)
+		require.Equal(t, client.DeleteRequests(expectedDeleteRequests), deleteRequests)
+	})
+
+	// Query lines
+	t.Run("verify query time filtering", func(t *testing.T) {
+		// reset boltdb-shipper client and restart querier
+		storage.ResetBoltDBIndexClientWithShipper()
+		require.NoError(t, tQuerier.Restart())
+
+		// update expectedStreams as per the issued requests
+		expectedStreams[0].Values = append(expectedStreams[0].Values[:1], expectedStreams[0].Values[2:]...)
+		expectedStreams[3].Values = expectedStreams[3].Values[2:]
+		expectedStreams = expectedStreams[:4]
+
+		// query and verify that we get the resp which matches expectedStreams
+		resp, err := cliQueryFrontend.RunRangeQuery(context.Background(), `{job="fake"}`)
+		require.NoError(t, err)
+
+		validateQueryResponse(resp)
 	})
 
 	// Wait until delete request is finished
 	t.Run("wait-until-delete-request-processed", func(t *testing.T) {
+		tenantLimits := tCompactor.GetTenantLimits(tenantID)
+		tenantLimits.DeletionMode = "filter-and-delete"
+		require.NoError(t, tCompactor.SetTenantLimits(tenantID, tenantLimits))
+
+		// all the delete requests should have been processed
```

Comment:
How do you know the delete requests have been processed?

Reply:
I am verifying that below in `require.Eventually`. Sorry, am I missing something?